src/sortStudents.ts

```diff
@@ -1,16 +1,72 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function countAverageGrade(studentGrades: number[]): number {
+  const sumGrades = studentGrades.reduce((sum: number, grade: number) => {
+    return sum + grade;
+  }, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sumGrades / studentGrades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  let studentsCopy = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      studentsCopy = studentsCopy.sort((student1, student2) => {
+        return order === 'asc'
+          ? student1[sortBy].localeCompare(student2[sortBy])
+          : student2[sortBy].localeCompare(student1[sortBy]);
+      });
+
+      break;
```

Comment:
Sort is a mutable method
```suggestion
      return studentsCopy.sort((student1, student2) => {
        return order === 'asc'
          ? student1[sortBy].localeCompare(student2[sortBy])
          : student2[sortBy].localeCompare(student1[sortBy]);
      });
```